### PR TITLE
Refactor to remove PHP 7 Error

### LINKED ISSUE #24841: Remove call to func_get_args() in WC_Order_Factory
**Is your feature request related to a problem? Please describe.**
There is a call to `func_get_args()` in `WC_Order_Factory::get_order()` that is causing a PHP 7.0 error notification in phpcs.

The feedback from phpcs is as follows:
```
FILE: woocommerce/includes/class-wc-order-factory.php
-----------------------------------------------------------------------------------------------
FOUND 1 ERROR AFFECTING 1 LINE
-----------------------------------------------------------------------------------------------
LINE 49: ERROR Since PHP 7.0, functions inspecting arguments, like func_get_args(), no longer report the original value as passed to a parameter, but will instead provide the current value. The parameter "$order_id" was changed on line 25. (PHPCompatibility.FunctionUse.ArgumentFunctionsReportCurrentValue.Changed)
```

**Describe the solution you'd like**
I would like for the call to `func_get_args()` to be removed. I can and am willing to submit a PR to fix this issue.

**Describe alternatives you've considered**
The only alternative I've been able to come up with is manually updating that file each time there is a WooCommerce update, either with a fix or an ignore comment for phpcs. Which is what I've done for a few of our clients.

### PR DESCRIPTION
**Pull request by jacobmc**
### All Submissions:

* [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Replaced call to func_get_args() in WC_Order_Factory::get_order() with a variable to store the original passed arguments. This removes the PHP 7.0 error described in issue #24841 

Closes #24841 .

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [ ] Have you successfully run tests with your changes locally?

### Changelog entry

> Refactored WC_Order_Factory::get_order() to remove function deprecated in PHP 7.0

### CODE REVIEW

**Comment by JacobMC** [comment]
For some reason, I was unable to get the unit tests and end to end testing working locally, so I was unable to run them.

File: includes/class-wc-order-factory.php [hunk 1]
```diff
@@ -46,7 +47,7 @@ public static function get_order( $order_id = false ) {
 		try {
 			return new $classname( $order_id );
 		} catch ( Exception $e ) {
-			wc_caught_exception( $e, __FUNCTION__, func_get_args() );
+			wc_caught_exception( $e, __FUNCTION__, $passed_args );
```
Comment (rodrigoprimo) [inline includes/class-wc-order-factory.php]: Maybe we don't need to preserve the old PHP < 7.0 behavior and pass the original value to `wc_caught_exception()`? I don't see a reason to pass the original value as the exception will be raised when trying to instantiate the class using the current value of `$order_id`. Or am I missing something?

I also think we should pass an array to `wc_caught_exception()` as this is what the function expects and also as `func_get_args()` returns an array.

To summarize, I believe we could simply change this line to:

```
 wc_caught_exception( $e, __FUNCTION__, array( $order_id ) ); 
```

And remove the line that sets `$passed_args`.

What do you think?

**Review by rodrigoprimo — COMMENTED** [review]
Thanks for your contribution, @JacobMC. I left a comment with a suggestion to simplify the code.

File: includes/class-wc-order-factory.php (hunk 1 above)
Comment (JacobMC) [reply includes/class-wc-order-factory.php]: That makes a lot of sense. I'll make those changes.

**Comment by JacobMC** [comment]
@rodrigoprimo thanks for the feedback! I implemented your suggestion.

**Comment by kloon** [comment]
@JacobMC @rodrigoprimo I believe just using `array( $order_id )` will give the same results as using `func_get_args` The whole reason for the error is because we are modifying $order_id in that function so the value passed is altered.

If we want to throw the exception with the original value passed we will need to store it in a variable first before assigning it again on line 25.

**Review by rodrigoprimo — APPROVED** [review]
Thanks for updating the PR, @JacobMC. Looks good to me now.

@kloon, I believe what you suggested is the same that I suggested and @JacobMC implemented? If that is not the case and I'm missing something, please let me know and I can create another PR to fix it. Or are you saying that you think we should raise the exception with the original `$order_id` value?
